Fix and export flexbox multiline-align-self test

sammygill opened the pull request:
#### c9a17e89b5093eaa55767477c32987eeaf3b43b5
<pre>
Fix and export flexbox multiline-align-self test
<a href="https://bugs.webkit.org/show_bug.cgi?id=296414">https://bugs.webkit.org/show_bug.cgi?id=296414</a>
<a href="https://rdar.apple.com/156555222">rdar://156555222</a>

Reviewed by Tim Nguyen.

It seems like this test was initially added in 97449@main as part of the
initial implementation of flex-wrap: wrap. It seems like Blink also has
had this test but updated the expectations over time for various
reasons. The test exercises the various values for align-self on flex
items in a multitude of contexts. Rather than going one by one, thinking
deeply about each subtest, and updating the expectations to what I think
are the right values, I am updating the expectations to match what Blink
currently has.

As part of this change, I also am exporting the test to WPT so that other
engines end up aware of any changes we make to the test and vice versa.
I also unrolled the generated content from the script and inserted the
resulting DOM tree in the test by just copying and pasting from the web
inspector. This will make it easier to view, debug, and make changes to
the test in the future. We could also break this test up into different
files for organizational purposes, but for now, just keep them in the same
file.

Canonical link: <a href="https://commits.webkit.org/297822@main">https://commits.webkit.org/297822@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/30461f312e765804982cc35cc2db30e703bfa57a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112982 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32717 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23195 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119186 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/63595 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33369 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41280 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85981 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/36675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115929 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26630 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101624 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66292 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19756 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62944 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19831 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122407 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40060 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29864 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94831 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40444 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97845 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94573 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24145 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39713 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17520 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36173 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39946 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39587 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42920 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41324 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5ed33b9d7e5922f3c7dbe4ff0be47b477f885569. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48460

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/30461f312e765804982cc35cc2db30e703bfa57a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48460

sammygill committed:
Fix and export flexbox multiline-align-self test
https://bugs.webkit.org/show_bug.cgi?id=296414
rdar://156555222

Reviewed by Tim Nguyen.

It seems like this test was initially added in 97449@main as part of the
initial implementation of flex-wrap: wrap. It seems like Blink also has
had this test but updated the expectations over time for various
reasons. The test exercises the various values for align-self on flex
items in a multitude of contexts. Rather than going one by one, thinking
deeply about each subtest, and updating the expectations to what I think
are the right values, I am updating the expectations to match what Blink
currently has.

As part of this change, I also am exporting the test to WPT so that other
engines end up aware of any changes we make to the test and vice versa.
I also unrolled the generated content from the script and inserted the
resulting DOM tree in the test by just copying and pasting from the web
inspector. This will make it easier to view, debug, and make changes to
the test in the future. We could also break this test up into different
files for organizational purposes, but for now, just keep them in the same
file.

Canonical link: https://commits.webkit.org/297822@main
